src/Details.jsx

```diff
@@ -1,8 +1,35 @@
-import { useParams} from 'react-router-dom'
+import { useParams } from 'react-router-dom'
+import { useQuery } from '@tanstack/react-query'
+import fetchPet from './fetchPet'
 
 const Details = () => {
   const { id } = useParams()
-  return <h2>{id}</h2>
+  // ["details", id] will be passed as the queryKey to fetchPet
+  const results = useQuery(['details', id], fetchPet)
```

Comment:
we will pass the queryKey to `useQuery` as an array of first parameter. This says `if there is cached data for details key got get it don't fetch` but if not then run fetchPet function.